src/mlpack/methods/ann/layer/adaptive_max_pooling.hpp

```diff
@@ -0,0 +1,167 @@
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ADAPTIVE_MAX_POOLING_HPP
+#define MLPACK_METHODS_ANN_LAYER_ADAPTIVE_MAX_POOLING_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the AdaptiveMaxPooling layer.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class AdaptiveMaxPooling
+{
+ public:
+  //! Create the AdaptiveMaxPooling object.
+  AdaptiveMaxPooling();
+
+  /**
+   * Create the AdaptiveMaxPooling object.
+   *
+   * @param outputWidth Width of the output.
+   * @param outputHeight Height of the output.
+   */
+  AdaptiveMaxPooling(const size_t outputWidth,
+                     const size_t outputHeight);
+
+  /**
+   * Create the AdaptiveMaxPooling object.
+   *
+   * @param outputShape A two-value tuple indicating width and height of the output.
+   */
+  AdaptiveMaxPooling(const std::tuple<size_t, size_t> outputShape);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename eT>
+  void Forward(const arma::Mat<eT>& input, arma::Mat<eT>& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, using 3rd-order tensors as
+   * input, calculating the function f(x) by propagating x backwards through f.
+   * Using the results from the feed forward pass.
+   *
+   * @param input The propagated input activation.
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename eT>
+  void Backward(const arma::Mat<eT>& input,
+                arma::Mat<eT>& gy,
+                arma::Mat<eT>& g);
+
+  //! Get the output parameter.
+  const OutputDataType& OutputParameter() const
+  { return poolingLayer.OutputParameter(); }
+
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return poolingLayer.OutputParameter(); }
+
+  //! Get the delta.
+  const OutputDataType& Delta() const { return poolingLayer.Delta(); }
+  //! Modify the delta.
+  OutputDataType& Delta() { return poolingLayer.Delta(); }
+
+  //! Get the width.
+  size_t InputWidth() const { return poolingLayer.InputWidth(); }
+  //! Modify the width.
+  size_t& InputWidth() { return poolingLayer.InputWidth(); }
+
+  //! Get the height.
```

Comment:
input height

Reply:
Really sorry I missed them. Making the changes now.